mp/importer/batcher.py

```diff
@@ -5,32 +5,45 @@
 The user can select the batch size, the maximum number of batches and the starting batch for yielding elements
 Other function will be added to randomly sample elements and to retrieve the last batch  
 """
-def run_in_batches(iterable, end_batch_callback=None, batch_size=2, batch_start=0, max_batches=None):	
+def run_in_batches(iterable, end_batch_callback=None, batch_size=2, batch_start=0, max_batches=None, seed=None, percentage=None):	
     """ Create batches from an iterable
    
     This function is a generator that creates batches of elements from an iterable 
     The callback function is called when a) maximum batches are reached or b) iterable is exhausted 
     All batches (complete and incomplete) are ended by a callback function
+
+    If random options are not None, gets a percentage of the total data
+    Pseudo-samples randomly the elements in the selected percentage (approximate)
+    The main script logs the seed to make it repeatable
     """ 	
+    import random
     c = 0
     ended = False
+    if percentage is not None:									
+        random.seed(seed)									# seed can be passed as kw to be repeatable
+        number = int(((100-percentage)*batch_size)/100)						# e.g. to yield the 30% of batch i mask the 70%
+        mask = set(random.sample(range(batch_size), number))										
+    else:
+        mask = []
```

Comment:
On Fri, Jun 05, 2015 at 01:40:54AM -0700, kiarasky wrote:
> >      c = 0
> >      ended = False
> > +    if percentage is not None:									
> > +        random.seed(seed)									# seed can be passed as kw to be repeatable
> > +        number = int(((100-percentage)*batch_size)/100)						# e.g. to yield the 30% of batch i mask the 70%
> > +        mask = set(random.sample(range(batch_size), number))										
> > +    else:
> > +        mask = []
> 
> Yes... however, if mask is None i cannot iterate over mask (position_in_batch in mask)... i'll see if there is a way to circumvent that.

if mask is not None and ...:
    ...apply_mask...
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/vanguardistas/mp.importer/pull/3/files#r31796039

-- 
Brian Sutherland